sdk/tests/js/tests/tex-image-and-sub-image-2d-with-image-bitmap-from-image.js

```diff
@@ -58,12 +60,18 @@ function generateTest(internalFormat, pixelFormat, pixelType, prologue, resource
         gl.clearColor(0,0,0,1);
         gl.clearDepth(1);
 
+        var testCanvas = document.createElement('canvas');
+        var ctx = testCanvas.getContext("2d");
+        setCanvasToRedGreen(ctx);
```

Comment:
For the construction of ImageBitmap from image, it would be better to create a resource .png with the required 4 colors, and load it with premultiplyAlpha={default,none} as well as colorspaceConversion=none. The reason is that every current canvas implementation uses a premultiplied alpha representation internally, which is lossy, and which is probably why the tolerances are so high.

In a follow-up pull request, could you construct such an image using the Gimp or Photoshop (let me know if you need help) and replace the canvas in both this and the 3d-with-image-bitmap-from-image test below with that? Thanks.